core/src/main/java/org/apache/carbondata/core/locks/CarbonLockUtil.java

```diff
@@ -107,4 +114,36 @@ public static int getLockProperty(String property, int defaultValue) {
     }
   }
 
+  /**
+   * Currently the segment lock files are not deleted immediately when unlock,
+   * so it needs to delete expired lock files before delete loads.
+   */
+  public static void deleteExpiredSegmentLockFiles(CarbonTable carbonTable) {
+    LoadMetadataDetails[] details =
```

Comment:
I think you can better list the segment lock files from locks folder and check the modified time of it and remove . Otherwise if you do as per tablestatus then namenode calls would be many.

Reply:
I think it is better to use this method to check the status and last modified time of the segment at the same time. As the size of the tablestatus file is getting larger, it is a problem, there are many places will scan this file, not just here, we need to solve this problem. I have raise a [topic](http://apache-carbondata-dev-mailing-list-archive.1130556.n5.nabble.com/The-size-of-the-tablestatus-file-is-getting-larger-does-it-impact-the-performance-of-reading-this-fi-td41941.html) to discuss this on mailling list.